include/bitcoin/protocol/zmq/message.hpp

```diff
@@ -66,12 +66,31 @@ class BCP_API message
 
     /// Remove a part from the queue top, false if empty queue or invalid.
     bool dequeue();
-    bool dequeue(uint32_t& value);
     bool dequeue(data_chunk& value);
     bool dequeue(std::string& value);
     bool dequeue(hash_digest& value);
     bool dequeue(address& value);
 
+    template<typename T,
+             typename std::enable_if<std::is_integral<T>::value, int>::type ...>
+    bool dequeue(T& value)
+    {
+        if (queue_.empty())
```

Comment:
This should probably have been `if (queue_.size() != 1)` but I haven't looked at other assumptions.

Reply:
Hrm, unless I'm missing something, I think that's incorrect.  If enqueue is called twice, the size check would fail on the first dequeue.  Empty appears to be the proper check in this case.

Reply:
Right, I was thinking of the number of elements in the first stack element (which is already guarded,so good). Better to have coffee first.